lib/fog/openstack/models/identity_v3/projects.rb

```diff
@@ -8,8 +8,20 @@ class V3
         class Projects < Fog::OpenStack::Collection
           model Fog::Identity::OpenStack::V3::Project
 
+          @@cache = {}
+          Fog::Identity::OpenStack::V3::Project.use_cache(@@cache)
+
           def all(options = {})
-            load_response(service.list_projects(options), 'projects')
+            cached_project, expires = @@cache[{token: service.auth_token, options: options}]
```

Comment:
hm this doesn't work, I think I am able to rescope the same token, so I am somehow getting here projects that I don't have rights to.

And later in the code getting 
Expected([201]) <=> Actual(401 Unauthorized)
       excon.error.response
         :body      => "{\"error\": {\"message\": \"Could not find project: 6923ecb4505e44a9b614912ce6c7cd5b-a3dd363c-de55-4984-9f30-eec3c5a (Disable debug mode to suppress these details.)\", \"code\": 401, \"title\": \"Unauthorized\"}}"